### PR TITLE
Refactoring GeometrySettings.h/.cpp and GeomUtils.h/.cpp

### PR DESCRIPTION
**Pull request by SamuilsRulovs**

### CODE REVIEW

File: Core/src/IfcGeometryConverter/GeomUtils.h [hunk 1]
```diff
@@ -62,93 +62,286 @@ namespace OpenInfraPlatform
...
 
+				/**********************************************************************************************/
+				/*! \brief Calculates a closest point on the line to the point. 
+				*
+				* \param[in] point					Point outside the line. 
+				* \param[in] line_origin			Line origin point. 
+				* \param[in] line_direction			Direction of the line. 	
+				*
+				* \param[out] closest				Closest point on the line to the point.
+				*/
```
Comment (SamuilsRulovs) [inline Core/src/IfcGeometryConverter/GeomUtils.h:226]: I am not sure, if this description is correct. Can you please give me a feedback regarding this doxy-comment

File: Core/src/IfcGeometryConverter/GeomUtils.h [hunk 2]
```diff
@@ -62,93 +62,286 @@ namespace OpenInfraPlatform
...
+				/*! \brief Verify if point is on the segment of the line. 
+				*
+				* \param[in] lambda					Lambda of the line. 
+				* \param[in] point					Point which should be verified. 
+				* \param[in] line_origin			Line origin point.
+				* \param[in] line_direction			Direction of the line.
+				*
+				* \return 							Returns true, if point is on the line. False otherwise.
+				*/
 				static bool isPointOnLineSegment(double& lambda,
```
Comment (SamuilsRulovs) [inline Core/src/IfcGeometryConverter/GeomUtils.h:259]: I am not sure, if this description is correct. Can you please give me a feedback regarding this doxy-comment

File: Core/src/IfcGeometryConverter/GeomUtils.h [hunk 3]
```diff
@@ -62,93 +62,286 @@ namespace OpenInfraPlatform
...
+				/**********************************************************************************************/
+				/*! \brief Extrude geometry. 
+				*
+				* \param[in] paths					List of several sets of points of the geometry.
+				* \param[in] dir					The direction, where geometry should be extruded.
+				* \param[in] closed					Boolean operator, which shows, if geometry is closed.
+				*
+				* \param[out] poly_data				List of faces of the geometry.
+				*/
 				static void extrude(const std::vector<std::vector<carve::geom::vector<2>>>& paths,
```
Comment (SamuilsRulovs) [inline Core/src/IfcGeometryConverter/GeomUtils.h:273]: I am not sure, if this description is correct. Can you please give me a feedback regarding this doxy-comment

File: Core/src/IfcGeometryConverter/GeomUtils.h [hunk 4]
```diff
@@ -62,93 +62,286 @@ namespace OpenInfraPlatform
...
 
+				/**********************************************************************************************/
+				/*! \brief Incorporates voids to the geometry. 
+				*
+				* \param[in] paths					List of several sets of points. 
+				*
+				* \param[out] merged_path			Arranged list of points.
+				* \param[out] triangulated			List of triangles to visualize geometry.
+				* \param[out] path_all_loops		List of several sets of indices. 
+				*/
```
Comment (SamuilsRulovs) [inline Core/src/IfcGeometryConverter/GeomUtils.h:286]: I am not sure, if this description is correct. Can you please give me a feedback regarding this doxy-comment

File: Core/src/IfcGeometryConverter/GeomUtils.h [hunk 5]
```diff
@@ -62,93 +62,286 @@ namespace OpenInfraPlatform
...
 
+				/**********************************************************************************************/
+				/*! \brief Corrects a winding of the points in curves.
+				*
+				* \param[in] face_loops_input		List of several sets of points.
+				*
+				* \param[out] normal_first_loop		Normalised first loop. 
+				*
+				* return							List of several sets of points with correct winding. 			
+				*/
```
Comment (SamuilsRulovs) [inline Core/src/IfcGeometryConverter/GeomUtils.h:300]: I am not sure, if this description is correct. Can you please give me a feedback regarding this doxy-comment

File: Core/src/IfcGeometryConverter/GeomUtils.h [hunk 6]
```diff
@@ -62,93 +62,286 @@ namespace OpenInfraPlatform
...
 
+				/**********************************************************************************************/
+				/*! \brief ???
+				*
+				* \param[in] eye					???
+				* \param[in] center					???
+				* \param[in] up						??? 
+				*
+				* \param[out] m						???
+				*/
```
Comment (SamuilsRulovs) [inline Core/src/IfcGeometryConverter/GeomUtils.h:313]: I do not know, how to describe this function. Can you please give me a hint?

File: Core/src/IfcGeometryConverter/GeomUtils.h [hunk 7]
```diff
@@ -62,93 +62,286 @@ namespace OpenInfraPlatform
...
 
+				/**********************************************************************************************/
+				/*! \brief Applies a position to the point.
+				*
+				* \param[in] listOfPoint				A list of points to be converted.
+				* \param[in] positionMatrix			A position matrix, which should be applied to the points.
+				* \param[in] ammountOfPoints			Number of points.
+				*
+				* \return[out]							List of points with applied position.
+				*/
```
Comment (SamuilsRulovs) [inline Core/src/IfcGeometryConverter/GeomUtils.h:369]: I forgot to correct this doxy-comment. I will correct it.

**Review by SamuilsRulovs — COMMENTED** [review]
I am not sure about correctness of several doxy-comments. Please correct me if necessary.

File: Core/src/IfcGeometryConverter/GeomUtils.cpp [hunk 8]
```diff
@@ -659,13 +665,14 @@ bool GeomUtils::isPointOnLineSegment( double& target_lambda,
...
-* \param[in] initialPointSeg2		Initial point of the second intersecting segment.			
+* \param[in] initialPointSeg2		Initial point of the second intersecting segment.
 * \param[in] terminalPointSeg2		Terminal point of the second intersecting segment.
-* \param[out] distToIntesection1	Distance from initialPointSeg1 to the segment intresection, which is calculated as section of the first segment. 
-* \param[out] distToIntesection2	Distance from initialPointSeg2 to the segment intresection, which is calculated as section of the second segment. 
+* \param[out] distToIntesection1	Distance from initialPointSeg1 to the segment intresection, which is calculated as section of the first segment.
+* \param[out] distToIntesection2	Distance from initialPointSeg2 to the segment intresection, which is calculated as section of the second segment.
+
 * \return							Returns true, if lines are intersecting with each other. False otherwise.
 */
```
Comment (pjanck) [inline Core/src/IfcGeometryConverter/GeomUtils.cpp]: this should probably be in header as well

File: Core/src/IfcGeometryConverter/GeomUtils.h [hunk 9]
```diff
@@ -62,93 +62,286 @@ namespace OpenInfraPlatform
...
-
+					const std::vector<carve::geom::vector<2>>& polygon);
+
+				/**********************************************************************************************/
+				/*! \brief Calculates coordinates of the intersection point.
+				*
+				* \param[in] initialPointSeg1		Initial point of the first intersecting segment.
+				* \param[in] terminalPointSeg1		Terminal point of the first intersecting segment.
+				* \param[in] initialPointSeg2		Initial point of the second intersecting segment, which describes the direction of the line.
+				* \param[in] terminalPointSeg2		Terminal point of the second intersecting segment, which describes the direction of the line.
```
Comment (pjanck) [inline Core/src/IfcGeometryConverter/GeomUtils.h]:
```suggestion
				* \param[in] terminalPointSeg2		2nd line segment, 2nd point.
```

Apply similarly for all comments of this sort: use (initial, terminal) for lines that are of fixed length, and (1st, 2nd) for lines that extend into infinity. 

I'd also include a comment in the description of the function, that the first/second line are to be taken as limited in length by the points or infinite.

File: Core/src/IfcGeometryConverter/GeomUtils.h [hunk 10]
```diff
@@ -62,93 +62,286 @@ namespace OpenInfraPlatform
...
+				* \return							Returns true, if lines are intersecting with each other. False otherwise.
+				*/
 				static bool GeomUtils::LineToLineIntersection(const carve::geom::vector<2>& initialPointSeg1,
 					const carve::geom::vector<2>& terminalPointSeg1,
 					const carve::geom::vector<2>& initialPointSeg2,
 					const carve::geom::vector<2>& terminalPointSeg2,
 					carve::geom::vector<2> & intersectionPoint);
 
+				/**********************************************************************************************/
+				/*! \brief Appedns vector of points (in 3D) to the curve.
```
Comment (pjanck) [inline Core/src/IfcGeometryConverter/GeomUtils.h]:
```suggestion
				/*! \brief Appends vector of points (in 3D) to the curve.
```

Include a note, that duplication of points is checked and duplicate points are skipped.

File: Core/src/IfcGeometryConverter/GeomUtils.h [hunk 11]
```diff
@@ -62,93 +62,286 @@ namespace OpenInfraPlatform
...
 					const carve::math::Matrix& matrix_a);
 
+				/**********************************************************************************************/
+				/*! \brief Calculates a closest point on the line to the point. 
+				*
+				* \param[in] point					Point outside the line. 
+				* \param[in] line_origin			Line origin point. 
+				* \param[in] line_direction			Direction of the line. 	
+				*
+				* \param[out] closest				Closest point on the line to the point.
```
Comment (pjanck) [inline Core/src/IfcGeometryConverter/GeomUtils.h]:
```suggestion
				/*! \brief Calculates the closest point on the line given another point. 
				*
				* \param[in] point					Point outside the line. 
				* \param[in] line_origin			Line origin point. 
				* \param[in] line_direction			Direction of the line. 	
				*
				* \param[out] closest				Closest point on the (infinite) line to the point.
```

File: Core/src/IfcGeometryConverter/GeomUtils.h (hunk 1 above)
Comment (pjanck) [reply Core/src/IfcGeometryConverter/GeomUtils.h:226]: See my suggestion. I believe that you've described it quite nicely, only a few corrections from my side to make it even more clear.

File: Core/src/IfcGeometryConverter/GeomUtils.h [hunk 12]
```diff
@@ -62,93 +62,286 @@ namespace OpenInfraPlatform
...
+				*/
 				static void closestPointOnLine(const buw::Vector3f& point,
 					const buw::Vector3f& line_origin,
 					const buw::Vector3f& line_direction,
 					buw::Vector3f& closest);
 
+				/**********************************************************************************************/
+				/*! \brief Verify if point is on the segment of the line. 
+				*
+				* \param[in] lambda					Lambda of the line.
```
Comment (pjanck) [inline Core/src/IfcGeometryConverter/GeomUtils.h]:
```suggestion
				* \param[out] lambda					The ratio of where the point lies on the line. 
```

File: Core/src/IfcGeometryConverter/GeomUtils.h [hunk 13]
```diff
@@ -62,93 +62,286 @@ namespace OpenInfraPlatform
...
 					buw::Vector3f& closest);
 
+				/**********************************************************************************************/
+				/*! \brief Verify if point is on the segment of the line. 
+				*
+				* \param[in] lambda					Lambda of the line. 
+				* \param[in] point					Point which should be verified. 
+				* \param[in] line_origin			Line origin point.
+				* \param[in] line_direction			Direction of the line.
+				*
```
Comment (pjanck) [inline Core/src/IfcGeometryConverter/GeomUtils.h:252]: I'd include an example to make this comment more understandable (I believe there should be an `\example` key for doxycomments): `lambda = 0` means the point equals `line_origin`, while `lambda=1` equals `line_origin + line_direction`.

File: Core/src/IfcGeometryConverter/GeomUtils.h [hunk 14]
```diff
@@ -62,93 +62,286 @@ namespace OpenInfraPlatform
...
 					const buw::Vector3f& line_origin,
 					const buw::Vector3f& line_direction);
-
+				
+				/**********************************************************************************************/
+				/*! \brief Extrude geometry. 
+				*
+				* \param[in] paths					List of several sets of points of the geometry.
+				* \param[in] dir					The direction, where geometry should be extruded.
+				* \param[in] closed					Boolean operator, which shows, if geometry is closed.
```
Comment (pjanck) [inline Core/src/IfcGeometryConverter/GeomUtils.h]:
```suggestion
				* \param[in] paths					List of several sets of points of the geometry (profiles to be extruded).
				* \param[in] dir					The direction + length, where geometry should be extruded.
				* \param[in] closed					Should the geometry be closed, i.e. include a face at the bottom and top as well or is only the mantel of interest?
```

File: Core/src/IfcGeometryConverter/GeomUtils.h (hunk 4 above)
Comment (pjanck) [reply Core/src/IfcGeometryConverter/GeomUtils.h:286]: We need to talk about this one.

File: Core/src/IfcGeometryConverter/GeomUtils.h (hunk 5 above)
Comment (pjanck) [reply Core/src/IfcGeometryConverter/GeomUtils.h:300]: Yeah, this one as well.

File: Core/src/IfcGeometryConverter/GeomUtils.h (hunk 6 above)
Comment (pjanck) [reply Core/src/IfcGeometryConverter/GeomUtils.h:313]: That's a bummer. I don't know that exactly either. We'll have to look where this function is being used, perhaps that will open our eyes.

**Review by pjanck — CHANGES_REQUESTED** [review]
Some suggestions for improvement.

File: Core/src/IfcGeometryConverter/GeomUtils.cpp (hunk 8 above)
Comment (SamuilsRulovs) [reply Core/src/IfcGeometryConverter/GeomUtils.cpp]: Resolved: 
40fe375

File: Core/src/IfcGeometryConverter/GeomUtils.h (hunk 9 above)
Comment (SamuilsRulovs) [reply Core/src/IfcGeometryConverter/GeomUtils.h]: Resolved: e82ca1a

File: Core/src/IfcGeometryConverter/GeomUtils.h (hunk 13 above)
Comment (SamuilsRulovs) [reply Core/src/IfcGeometryConverter/GeomUtils.h:252]: image

I guess \example is used in Doxygen a little bit differently. I will add this example with \note

File: Core/src/IfcGeometryConverter/GeomUtils.h (hunk 11 above)
Comment (SamuilsRulovs) [reply Core/src/IfcGeometryConverter/GeomUtils.h]: Resolved: 36ad1fc

File: Core/src/IfcGeometryConverter/GeomUtils.h (hunk 12 above)
Comment (SamuilsRulovs) [reply Core/src/IfcGeometryConverter/GeomUtils.h]: Resolved: 36ad1fc

File: Core/src/IfcGeometryConverter/GeomUtils.h (hunk 14 above)
Comment (SamuilsRulovs) [reply Core/src/IfcGeometryConverter/GeomUtils.h]: Resolved: 36ad1fc

File: Core/src/IfcGeometryConverter/GeomUtils.h (hunk 13 above)
Comment (SamuilsRulovs) [reply Core/src/IfcGeometryConverter/GeomUtils.h:252]: Resolved: 36ad1fc

File: Core/src/IfcGeometryConverter/GeomUtils.h (hunk 10 above)
Comment (SamuilsRulovs) [reply Core/src/IfcGeometryConverter/GeomUtils.h]: Resolved in: 
36ad1fc
51d6064

File: Core/src/IfcGeometryConverter/GeomUtils.h (hunk 13 above)
Comment (pjanck) [reply Core/src/IfcGeometryConverter/GeomUtils.h:252]: note is fine as well!

**Review by pjanck — APPROVED** [review]
We can leave the open comments for another time.

**Pull request merged**